build.sh

```diff
@@ -22,6 +23,24 @@ output_path=$(realpath -m "${1}")
 shim_path=$(realpath -m "${2}")
 rootfs_dir=$(realpath -m "${3}")
 
+if [ "${args['luks']}" = 'true' ]; then
+  printf "Enter the LUKS2 password for the image: "
+  read PASSWD
+  luks_enabled=true
+  if [ "${args['arch']}" = 'arm64' ]; then
+    CRYPTSETUP=cryptsetup_arm64
+  else
+    CRYPTSETUP=cryptsetup_x86_64
+  fi
+  CRYPT_PATH=$(realpath -m bootloader/bin/cryptsetup)
+  if [ ! -f $CRYPT_PATH ]; then
+    print_info "downloading cryptsetup binary"
+    curl -LO "https://github.com/FWSmasher/CryptoSmite/raw/main/${CRYPTSETUP}" 
```

Comment:
Use wget for downloading files. Check how it's used in other parts of the built script.

Also, don't download the cryptsetup binary from an untrusted third party repository. The https://github.com/ading2210/shimboot-binaries repo exists for this purpose instead.